partitura/performance.py

```diff
@@ -289,6 +335,68 @@ def __init__(
         self.lyricist = lyricist
         self.copyright = copyright
 
+        if ensure_unique_tracks:
+            self.sanitize_track_numbers()
+
+    @property
+    def num_tracks(self) -> int:
+        """
+        Number of tracks in the performance
+        """
+        n_tracks = len(
+            set(
+                [(i, n.get("track", -1)) for i, pp in enumerate(self) for n in pp.notes]
+                + [
+                    (i, c.get("track", -1))
+                    for i, pp in enumerate(self)
+                    for c in pp.controls
+                ]
+                + [
+                    (i, p.get("track", -1))
+                    for i, pp in enumerate(self)
+                    for p in pp.programs
+                ]
+            )
+        )
+
+        return n_tracks
+
+    def sanitize_track_numbers(self) -> None:
```

Comment:
what is the use of this function? afaik partitura currently only loads into performances via load_performance_midi and load_match, in both cases only a single performedpart is in the performance. multiple performedpart objects are allowed for similarity with the score object and such that multiple midi files (e.g. many instruments) can be represented in a single object (akin to the score object for an orchestral piece). in this case the track numbers are not assumed to be unique per performedpart/midi file/instrument.

Reply:
Since `PerformedPart` objects can be created directly from note arrays, it would be possible to have multiple parts in the same performance (e.g., generating a bass part + a melody part). But if not  explicitly stated, the track information would be the same for each `PerformedPart` in the `Performance`. This method avoids would "sanitize" this issue, so that the track information in each of the `PerformedPart` objects does not overlap.

Reply:
Ah so that's a helper for manual creation. Can we add a safety against too many tracks? only 16 are allowed I think.

Reply:
I think the limit is only in the channels, AFAIK, not on the tracks?

Reply:
ah yes, that could be.

Reply:
Is this resolved?

Reply:
Yes